algosdk/abi.py

```diff
@@ -0,0 +1,570 @@
+        return Value.make_uint(value, 32)
+
+    @staticmethod
+    def make_uint64(value):
+        return Value.make_uint(value, 64)
+
+    @staticmethod
+    def make_ufixed(value, size, precision):
+        assert isinstance(value, int)
+        ufixed_value_type = Type.make_ufixed_type(size, precision)
+        uint_value = Value.make_uint(value, size)
+        uint_value.abi_type = ufixed_value_type
+        return uint_value
+
+    @staticmethod
+    def make_bool(value):
+        assert isinstance(value, bool)
+        return Value(Type.make_bool_type, value)
+
+    @staticmethod
+    def make_string(value):
+        assert isinstance(value, str)
+        return Value(Type.make_string_type, value)
+
+    @staticmethod
+    def make_byte(value):
+        assert isinstance(value, bytes)
+        return Value(Type.make_byte_type, value)
+
+    @staticmethod
+    def make_address(value):
+        assert len(value) == 32
+        return Value(Type.make_address_type, value)
+
+    @staticmethod
+    def make_dynamic_array(values, element_type):
+        assert isinstance(values, list())
+        assert isinstance(element_type, Type)
+        if len(values) >= (2 ** 16):
+            raise error.ABIValueError(
+                f"dynamic array values length: {len(values)} must be less than uint16"
+            )
+        for val in values:
+            assert isinstance(val, Value)
+            if val.abi_type != element_type:
+                raise error.ABIValueError(
+                    f"dynamic array type mismatch: {val.abi_type} != {element_type}"
+                )
+        return Value(Type.make_dynamic_array_type(element_type), values)
+
+    @staticmethod
+    def make_static_array(values):
+        assert isinstance(values, list())
+        if len(values) >= (2 ** 16) or len(values) == 0:
+            raise error.ABIValueError(
+                f"dynamic array values length: {len(values)} must be greater than 0 and less than uint16"
+            )
+        for val in values:
+            assert isinstance(val, Value)
+            if val.abi_type != values[0].abi_type:
+                raise error.ABIValueError(
+                    f"dynamic array type mismatch: {val.abi_type} != {values[0].abi_type}"
+                )
+        return Value(
+            Type.make_static_array_type(values[0].abi_type, len(values)),
+            values,
+        )
+
+    @staticmethod
+    def make_tuple(values):
+        assert isinstance(values, list())
+        if len(values) >= (2 ** 16):
+            raise error.ABIValueError(
+                f"tuple values length: {len(values)} must be less than uint16"
+            )
+        tuple_types = []
+        for val in values:
+            assert isinstance(val, Value)
+            tuple_types.append(val.abi_type)
+
+        return Value(Type.make_tuple_type(tuple_types), values)
+
+    def get_uint(self):
+        if not isinstance(self.abi_type.abi_type_id, BaseType.Uint):
```

Comment:
```suggestion
        if self.abi_type.abi_type_id != BaseType.Uint:
```
I think you mean this?